src/kernel/tracelog.c

```diff
@@ -75,7 +75,7 @@ build_assert((TRACELOG_ENTRY_SIZE & (TRACELOG_ENTRY_SIZE - 1)) == 0);
 static inline void schedule_collator(void)
 {
     if (!atomic_swap_boolean(&tracelog.collator_scheduled, true))
-        enqueue_irqsafe(runqueue, &tracelog.collator);
+        async_apply((thunk)&tracelog.collator);
```

Comment:
This is called (indirectly) by tprintf(), which is called by the various debug prints when any of the `*_DEBUG` defines is enabled. Some of these debug prints are done in interrupt context (e.g. those in src/*/interrupt.c). So we should use async_apply_bh here.